feat(ast): Label AST fields with #[ts]

ottomated opened the pull request:
I think I found all of the AST fields that are typescript-only but I could be wrong. I did label decorators as `#[ts]` but not sure if that's correct.

ottomated commented:
* **#6987** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/6987?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @ottomated and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="11px" height="11px"/> <b>Graphite</b></a>
  <!-- Current dependencies on/for this PR: -->

DonIsaac commented on crates/oxc_ast/src/ast/js.rs:
````diff
@@ -1740,10 +1756,12 @@ pub struct Class<'a> {
     /// class Foo {}
     /// ```
+    #[ts]
     pub decorators: Vec<'a, Decorator<'a>>,
````
May not be ts, could be stage 3 decorators

overlookmotel replied:
Acorn [does not seem to support decorators](https://ast.sxzz.moe/#eNolyTEKwzAMQNGrGE0t5ASZCr1C6aRFqCqkOJKxnEIwvnsUsn3+65Bhhh/9ybkupcEEJQaxVY3m6MfX7HZH5Uzu6Zn6wJMsqKOmhCC80luqL6YIc4xMTbwhTJe7bZXltRe5eLXPlqNRB4wDeOQp3w==). So I think we can consider them as TS (for now).

DonIsaac commented on crates/oxc_ast/src/ast/js.rs:
```diff
@@ -2116,6 +2150,7 @@ pub struct AccessorProperty<'a> {
     /// Decorators applied to the accessor property.
     ///
     /// See [`Decorator`] for more information.
+    #[ts]
```
I think accessor properties themselves are ts-only, but I'm not certain

overlookmotel replied:
Yes, I believe that's correct.

Dunqing replied:
No, it's a JavaScript feature
https://github.com/tc39/proposal-decorators?tab=readme-ov-file#new-class-elements

overlookmotel replied:
Well that depends on whether we consider decorators to be JS or TS. As discussed in other comments here, I think we decided to treat them as TS for now.

I don't think it matters so much either way. It'll be easy to change later on - we just remove the `#[ts]` attrs from decorator-related fields, and the code will update. Such is the joy of codegen!

overlookmotel commented on crates/oxc_ast/src/ast/js.rs:
```diff
@@ -1441,6 +1446,7 @@ pub struct BindingPattern<'a> {
     pub kind: BindingPatternKind<'a>,
+    #[ts]
     pub type_annotation: Option<Box<'a, TSTypeAnnotation<'a>>>,
     pub optional: bool,
```
Looks like it's TS to me. Can't find [any `optional` fields in Acorn AST](https://ast.sxzz.moe/#eNolyrEOgzAMBNBfiTyzsFL1L6oumMFyPVAFO3ICSoX491piuqe7OyHDBF86qLKvpcEAJQpicw1z+CBP/YHa0zONkXNfQvO4hONicTlRU0IQ3ugtXldThCmKTE1qQxjuvdruLK9fkXve7LPnMOoF1x+GhCuR).

Boshen commented:
### Merge activity

* **Oct 29, 11:30 PM EDT**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc) once it meets the requirements.

ottomated committed:
feat(ast): Label AST fields with #[ts] (#6987)

I think I found all of the AST fields that are typescript-only but I could be wrong. I did label decorators as `#[ts]` but not sure if that's correct.
